src/Symfony/Component/CssSelector/CssSelector.php

```diff
@@ -82,35 +75,32 @@ class CssSelector
      */
     public static function toXPath($cssExpr, $prefix = 'descendant-or-self::')
     {
-        $translator = new Translator();
-
-        if (self::$html) {
-            $translator->registerExtension(new HtmlExtension($translator));
-        }
+        $converter = new Converter(self::$html);
 
-        $translator
-            ->registerParserShortcut(new EmptyStringParser())
-            ->registerParserShortcut(new ElementParser())
-            ->registerParserShortcut(new ClassParser())
-            ->registerParserShortcut(new HashParser())
-        ;
-
-        return $translator->cssToXPath($cssExpr, $prefix);
+        return $converter->toXPath($cssExpr, $prefix);
     }
 
     /**
      * Enables the HTML extension.
+     *
+     * @deprecated as of 2.8, will be removed in 3.0. Use the \Symfony\Component\CssSelector\Converter class instead.
      */
     public static function enableHtmlExtension()
     {
+        @trigger_error('The method '.__METHOD__.' is deprecated as of 2.8 and will be removed in 3.0. Use \Symfony\Component\CssSelector\Converter instead.', E_USER_DEPRECATED);
```

Comment:
And I would change "Use \Symfony\Component\CssSelector\Converter instead." to "Use the \Symfony\Component\CssSelector\Converter class instead." (same as in the docblock).